h2o-core/src/main/java/hex/FeatureInteractions.java

```diff
@@ -0,0 +1,307 @@
+package hex;
+
+import hex.genmodel.algos.tree.SharedTreeNode;
+import org.apache.commons.lang.mutable.MutableInt;
+import water.util.TwoDimTable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class FeatureInteractions {
```

Comment:
would love to see some unit tests for this class

Reply:
1e91e7a